### DISCUSSION TITLE
how to reject a new peer stream

### DISCUSSION THREAD

**Discussion by fofewyoung** [microsoft/msquic#3259] [Q&A]
```
case QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED:
            {
                     //  no SetCallbackHandler

                     // close here
                    MsQuic->StreamClose(Event->PEER_STREAM_STARTED.Stream);

                    // return success?
                    return QUIC_STATUS_SUCCESS;
            }
```
is it correct?

**Comment by nibanks** [comment]
You should explicitly shutdown (abort) the stream with an app-specific error code so the peer know why you did it. If you just call close, the stack defaults to error 0, which might not be what you want.

**Reply by fofewyoung** [reply]
in my case, the error code is not important.
i expect it should be closed as soon as possible.

**Comment by fofewyoung** [comment]
In fact, I want to do this because I'm afraid that aggressive clients will create a large number of streams, which will cause the server memory to crash; So I want to limit the number of streams created

**Reply by chunlei-wu** [reply]
Maybe, you can use the setting, peerBidiStreamCount, to limit the stream count.

**Reply by nibanks** [reply]
Yes, ideally, you should only configure the number of streams you're willing to accept at a time. That way, you don't need to reject anything because they can't even send too many to begin with.
